llm/bill_on_document_created.py

```diff
@@ -0,0 +1,332 @@
+from firebase_functions.firestore_fn import (
+    on_document_created,
+    Event,
+    DocumentSnapshot,
+)
+from llm_functions import get_summary_api_function, get_tags_api_function_v2
+from typing import TypedDict, NewType
+from collections import deque
+
+Category = NewType("Category", str)
+
+
+# This allows us to type the return of `get_topics`
+class TopicAndCategory(TypedDict):
+    # We use the name `tag` in Python, but `topic` in the database
+    topic: str
+    # Topic can be mapped directly to a category
+    category: Category
+
+
+# Using a 'deque' because appending to lists in Python is slow
+def get_categories_from_topics(
+    topics: list[str], topic_to_category: dict[str, Category]
+) -> deque[TopicAndCategory]:
+    to_return: deque[TopicAndCategory] = deque()
+    for topic in topics:
+        if topic_to_category.get(topic):
+            to_return.append(
+                TopicAndCategory(topic=topic, category=topic_to_category[topic])
+            )
+    return to_return
+
+
+@on_document_created(document="generalCourts/{session_id}/bills/{bill_id}")
+def add_summary_on_document_created(event: Event[DocumentSnapshot | None]) -> None:
```

Comment:
This `| None` came from `mypy` so I am hoping the typing information is coming from `@on_document_created`